framework/InputTemplates/TemplateBaseClass.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on February 22 2019
+@author: talbpaul
+
+This module is the base class for Input Templates, which use an established input
+template as an accelerated way to write new RAVEN workflows. Other templates
+can inherit from this base class for specific applications.
+"""
+from __future__ import division, print_function, unicode_literals, absolute_import
+import warnings
+warnings.simplefilter('default', DeprecationWarning)
+# standard library
+import os
+import sys
+import copy
+# external libraries
+# RAVEN libraries
+frameworkDir = os.path.normpath(os.path.join(os.path.dirname(__file__), '..'))
+sys.path.append(frameworkDir)
+from utils import xmlUtils
+
+
+class Template(object):
+  """
+    Generic class for templating input files.
+    Intended to be used to read a template, be given instructions on how to fill it,
+    and create a new set of input files.
+  """
+  # generic class members
```

Comment:
I can remove this comment, if that would make it clearer... things defined outside function definitions are class members instead of instance members, right? I'm not sure how else to call them, and this doesn't seem like an apt place to include a Python Class tutorial.

Reply:
I mean to explain how this container is used... not the Python meaning...

Reply:
Oh, sure, that comment is the next line down. I will add more explanation.